articles/changes.md

```diff
@@ -0,0 +1,245 @@
+
+In ROS 2 the CMake API was restructured to support this use case.
+
+
+### Minimal support for packages without manifest
+
+In ROS 1 only packages with a manifest file are considered by the build system.
+In ROS 2 it is possible to detect packages with supported build system in folders without a manifest file.
+If the package follows common practice it might even be possible to detect some of the missing meta information (like dependencies).
+
+
+## Messages, Services
+
+### Separated namespaces
+
+In ROS 1 .msg and .srv files can have the same name but the generated code collides.
+The same is the case for the request and response parts of services.
+
+In ROS 2 the generated code uses separate namespaces to guarantee it is collision-free.
+
+
+### Optional default values in message definitions
+
+In ROS 2 primitive values in messages can now have default values, set when the message is constructed.
+Default values for non-primitive fields, i.e. nesting messages, is not yet possible.
+
+
+### Optional upper bounds for arrays and strings
+
+This is necessary in order to calculate the maximum size of a message in memory, which allows for preallocation of messages with a dynamic sizes.
+This is useful for performance and for use cases like real-time.
+
+
+### Unify duration and time types
+
+In ROS 1 the duration and time types are defined in the client libraries.
+The member names of the data structures are different in C++ (sec, nsec) and Python (secs, nsecs).
+
+In ROS 2 these types will be defines as messages and therefore be consistent across languages.
+
+
+### Remove sequence field from Header message
+
+The field has been deprecated for a long time and was not set consistently in ROS 1.
+
+
+## Client libraries
+
+### Across languages
+
+#### Notifications
+
+In ROS 1 all information about the ROS graph must be polled from the master.
+In ROS 2 changes will be published instead.
+
+
+#### Components with life cycle
+
+In ROS 1 every node usually has its own main function.
+In ROS 2 it will be recommended to subclass from a component which has a life cycle.
+
+The life cycle can be used by tools like roslaunch to start a system composed of many components in a deterministic way.
+
+
+#### Parameters and Dynamic Reconfigure
+
+In ROS 1 global parameters and node-specific dynamic reconfigure parameters are two separate concepts.
+In ROS 2 a unified approach is being used.
+It will be similar to dynamic reconfigure and a node named "global parameter server" will accept requests to set values unconditionally.
+In ROS 1 all this information needs to be polled for changes in ROS 2 changes will be published to notify other entities.
+
+
+#### Threading model
+
+In ROS 1 the developer can only choose between single-threaded execution or multi-threaded execution.
+In ROS 2 more granular execution models are available.
```

Comment:
I don't know if this is accurate or maybe it's not what we should be emphasizing. Perhaps instead point out that it separates execution from nodes, allows execution over more than one node at a time (similar to what you said here, but not exactly the same), or that it is extensible whereas `ros::spin()`/`ros::AsynSpinner` wasn't.